Gate scrub leaves agent CLI binaries on PATH — auth check disagrees between dev and CI

## What happened

CLAUDE.md states:

> \`make gate\` runs in a scrubbed environment: agent credentials, **CLI auth state**, and dotenv autoload inputs are stripped before tests execute.

The implementation in \`tests/conftest.py::_scrub_agent_credentials_for_gate\` does two things:

1. Strips env vars listed in \`SCRUBBED_ENV_VARS\` (API keys etc.)
2. Remaps \`HOME\` to a tempdir so files like \`~/.claude/.credentials.json\` are not visible

It does **not** modify \`PATH\`. So agent CLI binaries (\`claude\`, \`codex\`, \`gemini\`, \`npx\`) installed in standard locations (\`/usr/local/bin\`, \`~/.npm-global/bin\`, \`~/.local/bin\`) remain discoverable.

\`check_agent_auth\` in \`config/auth.py\` for CLI-profile agents resolves auth as:

\`\`\`python
ok = shutil.which(profile.cli) is not None
\`\`\`

i.e. \"is the binary on PATH\" — not \"is the CLI actually authenticated.\" The result diverges between environments:

- **Dev machine with Claude CLI installed**: \`shutil.which(\"claude\")\` → path → \`_has_auth\` returns True → assignment takes the primary code path
- **CI runner with no agent CLIs installed**: \`shutil.which(\"claude\")\` → None → \`_has_auth\` returns False → assignment takes the no-auth fallback path, which produces a different routing-rationale string

Concrete evidence: PR #1111 (story #1102) failed CI deterministically across Python 3.11/3.12/3.13 with:

\`\`\`
AssertionError: assert 'complexity score 6 (MEDIUM) → tier mid (fallback: cheapest at floor tier, no auth)'
              == 'complexity score 6 (MEDIUM) → tier mid'
\`\`\`

Same commit, same test, passed on dev machine in a fully scrubbed env (\`env -i HOME=$HOME PATH=$PATH make gate\`) because the dev machine had Claude CLI on PATH.

## What was expected

The gate scrub should make agent CLI binaries undiscoverable so the auth check produces the same answer on dev and CI. After this lands:

- A test exercising assignment that does not explicitly mock \`_has_auth\` sees the same code path on dev and CI
- The recurring \"passes locally, fails CI\" failure mode for assignment-related tests stops
- CLAUDE.md's claim that \"CLI auth state is stripped\" becomes true in practice, not just for env-and-HOME-based config

## Notes

- The fix probably belongs in \`tests/conftest.py::_scrub_agent_credentials_for_gate\` — same fixture that already remaps HOME. Options: prepend a tempdir of stub binaries that don't exist; rebuild PATH from a minimal whitelist; \`monkeypatch.setattr(\"shutil.which\", ...)\` for the CLI names of interest.
- Related but distinct: the broader question of whether \`check_agent_auth\` should verify actual auth (e.g. \`claude --version\` returning 0 with valid token) rather than just binary presence. That's a deeper change; this story is about the scrub gap specifically.
- This explains the recurring class of \"works on my machine\" CI failures around assignment-related tests; future tests that touch \`_has_auth\` should not need to remember to mock it.

--- a/src/theforge/config/__init__.py
+++ b/src/theforge/config/__init__.py
@@ -19,7 +19,7 @@ from .defaults import (
     SUPPORTED_CLIS,
     generate_default_config,
 )
-from .gate_scrub import SCRUBBED_ENV_VARS, SCRUBBED_HOME_PATHS
+from .gate_scrub import SCRUBBED_CLI_LAUNCHERS, SCRUBBED_ENV_VARS, SCRUBBED_HOME_PATHS
 from .load import _validate_plan_provider, load_config
 from .models import (
     AGENT_REGISTRY,
@@ -100,6 +100,7 @@ __all__ = [
     "PROVIDER_SDK_MAP",
     "SUPPORTED_CLIS",
     "generate_default_config",
+    "SCRUBBED_CLI_LAUNCHERS",
     "SCRUBBED_ENV_VARS",
     "SCRUBBED_HOME_PATHS",
     # load

--- a/src/theforge/config/gate_scrub.py
+++ b/src/theforge/config/gate_scrub.py
@@ -5,6 +5,7 @@ from __future__ import annotations
 from pathlib import Path
 
 from .auth import PROVIDER_API_KEY_MAP
+from .defaults import SUPPORTED_CLIS
 from .models import AGENT_REGISTRY
 
 _PROVIDER_ENV_ALIASES: dict[str, tuple[str, ...]] = {
@@ -39,4 +40,6 @@ SCRUBBED_HOME_PATHS: tuple[Path, ...] = (
     Path(".gemini"),
 )
 
-__all__ = ["SCRUBBED_ENV_VARS", "SCRUBBED_HOME_PATHS"]
+SCRUBBED_CLI_LAUNCHERS: tuple[str, ...] = tuple(sorted({*SUPPORTED_CLIS, "npx"}))
+
+__all__ = ["SCRUBBED_CLI_LAUNCHERS", "SCRUBBED_ENV_VARS", "SCRUBBED_HOME_PATHS"]

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -4,6 +4,7 @@ from __future__ import annotations
 
 import ipaddress
 import os
+import shutil
 import socket as _socket_module
 import tempfile
 from pathlib import Path
@@ -11,7 +12,12 @@ from unittest.mock import patch
 
 import pytest
 
-from theforge.config import SCRUBBED_ENV_VARS, SCRUBBED_HOME_PATHS, ModelProfile
+from theforge.config import (
+    SCRUBBED_CLI_LAUNCHERS,
+    SCRUBBED_ENV_VARS,
+    SCRUBBED_HOME_PATHS,
+    ModelProfile,
+)
 
 # ---------------------------------------------------------------------------
 # Global network guard — installed at conftest load time
@@ -79,6 +85,9 @@ def _gate_scrub_enabled() -> bool:
     return os.environ.get("THEFORGE_ALLOW_AGENT_CREDS") != "1"
 
 
+_REAL_WHICH = shutil.which
+
+
 @pytest.fixture(scope="session", autouse=True)
 def _scrub_agent_credentials_for_gate():
     """Strip agent credentials/auth state unless explicitly opted out.
@@ -93,6 +102,12 @@ def _scrub_agent_credentials_for_gate():
     stripped = sorted(name for name in SCRUBBED_ENV_VARS if name in os.environ)
     original_env = {name: os.environ.get(name) for name in SCRUBBED_ENV_VARS}
     original_home = os.environ.get("HOME")
+    scrubbed_launchers = frozenset(SCRUBBED_CLI_LAUNCHERS)
+
+    def _scrubbed_which(cmd, mode: int = os.F_OK | os.X_OK, path: str | None = None):
+        if os.path.basename(os.fsdecode(cmd)) in scrubbed_launchers:
+            return None
+        return _REAL_WHICH(cmd, mode=mode, path=path)
 
     for name in SCRUBBED_ENV_VARS:
         os.environ.pop(name, None)
@@ -112,18 +127,19 @@ def _scrub_agent_credentials_for_gate():
             "[theforge-test-guard] scrubbed agent credentials: "
             + (", ".join(stripped) if stripped else "none")
         )
-        try:
-            yield
-        finally:
-            if original_home is None:
-                os.environ.pop("HOME", None)
-            else:
-                os.environ["HOME"] = original_home
-            for name, value in original_env.items():
-                if value is None:
-                    os.environ.pop(name, None)
+        with patch("shutil.which", side_effect=_scrubbed_which):
+            try:
+                yield
+            finally:
+                if original_home is None:
+                    os.environ.pop("HOME", None)
                 else:
-                    os.environ[name] = value
+                    os.environ["HOME"] = original_home
+                for name, value in original_env.items():
+                    if value is None:
+                        os.environ.pop(name, None)
+                    else:
+                        os.environ[name] = value
 
 
 @pytest.fixture(autouse=True)

--- a/tests/test_config_models.py
+++ b/tests/test_config_models.py
@@ -232,7 +232,8 @@ class TestModelsKeyConfig:
             },
             tmp_path,
         )
-        config = load_config(config_path)
+        with patch("theforge.config.load.check_agent_auth", return_value=(True, "")):
+            config = load_config(config_path)
         assert config.assignment.enabled is True
         assert len(config.agents) == 3
         tiers = {a.tier for a in config.agents}

--- a/tests/test_gate_scrub.py
+++ b/tests/test_gate_scrub.py
@@ -5,8 +5,19 @@ from pathlib import Path
 
 import pytest
 
-from theforge.config import SCRUBBED_ENV_VARS, SCRUBBED_HOME_PATHS
-from theforge.config.gate_scrub import SCRUBBED_ENV_VARS as MODULE_ENV_VARS
+from theforge.config import (
+    SCRUBBED_CLI_LAUNCHERS,
+    SCRUBBED_ENV_VARS,
+    SCRUBBED_HOME_PATHS,
+    ModelProfile,
+)
+from theforge.config.auth import check_agent_auth
+from theforge.config.gate_scrub import (
+    SCRUBBED_CLI_LAUNCHERS as MODULE_CLI_LAUNCHERS,
+)
+from theforge.config.gate_scrub import (
+    SCRUBBED_ENV_VARS as MODULE_ENV_VARS,
+)
 
 
 def test_scrubbed_env_vars_cover_expected_credentials():
@@ -34,11 +45,44 @@ def test_scrubbed_home_paths_cover_cli_auth_locations():
     assert Path(".gemini") in scrubbed
 
 
+def test_scrubbed_cli_launchers_cover_supported_auth_probes():
+    expected = {"claude", "codex", "gemini", "npx"}
+    assert expected == set(SCRUBBED_CLI_LAUNCHERS)
+    assert SCRUBBED_CLI_LAUNCHERS == MODULE_CLI_LAUNCHERS
+
+
 def test_gate_scrub_fixture_sets_dotenv_disable_and_rehomes_home():
     assert os.environ["PYTHON_DOTENV_DISABLED"] == "1"
     scrubbed_home = Path(os.environ["HOME"])
     for rel_path in SCRUBBED_HOME_PATHS:
         assert (scrubbed_home / rel_path).exists()
+
+
+def test_gate_scrub_fixture_hides_cli_auth_launchers():
+    claude_profile = ModelProfile(
+        name="dev",
+        cli="claude",
+        model="sonnet",
+        budget_usd=1.0,
+        timeout_seconds=300,
+        allowed_tools=(),
+    )
+    codex_profile = ModelProfile(
+        name="dev",
+        cli="codex",
+        model="gpt-5.4",
+        budget_usd=1.0,
+        timeout_seconds=300,
+        allowed_tools=(),
+    )
+
+    ok, reason = check_agent_auth(claude_profile, {})
+    assert ok is False
+    assert reason == "'claude' not found in PATH"
+
+    ok, reason = check_agent_auth(codex_profile, {})
+    assert ok is False
+    assert reason == "npx not found in PATH"
 
 
 def test_real_runner_sentinel_message_is_gate_specific():

